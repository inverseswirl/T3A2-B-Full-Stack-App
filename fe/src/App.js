import React from "react";
import "./App.css";
// import Login from "./pages/Login";
import Signup from "./pages/Signup";
import { Routes, Route, Link } from "react-router-dom";
import Mainhome from "./pages/Home";

function App() {
  return (
    <>
      <Routes>
        <Route path="/" element={<Mainhome />} />
        {/* <Route path="/login" element={<Login />} /> */}
        <Route path="/signup" element={<Signup />} />
      </Routes>
    </>
  );
}

export default App;
