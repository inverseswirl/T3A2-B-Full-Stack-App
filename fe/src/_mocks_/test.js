function fetch_data() {
  return fetch("https://secret-forest-05738.herokuapp.com/exercises");
}

function fetch_workout() {
  return fetch("https://secret-forest-05738.herokuapp.com/workouts");
}
const mock_obj = {
  exerciseId: 1,
  exercise: "Abs",
  activity: "Crunch",
  weights: 10,
  reps: 12,
};

module.exports = {
  fetch_data,
  mock_obj: mock_obj,
  fetch_workout,
};
