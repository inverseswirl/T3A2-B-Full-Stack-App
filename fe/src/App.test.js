import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { render, screen, fireEvent, getByText } from "@testing-library/react";
import Navbar from "./components/navbar";
import { fetch_data, fetch_workout, mock_obj } from "./_mocks_/test";

test("renders the Xero fitness brand name", () => {
  render(<Navbar />);
  const element = screen.getByTestId("brandname");
  expect(element).toHaveTextContent("Xero fitness");
});

test("fetches exercise list", async () => {
  expect.assertions(1);

  const results = await fetch_data();
  expect(results).not.toBeUndefined();
});
test("handles error for workout data fetching", async () => {
      await expect(fetch_workout()).rejects.toMatch('error');
 
});

test("object is stringified", () => {
  const json_data = JSON.stringify(mock_obj);
  expect(json_data).toBe(
    `{"exerciseId":1,"exercise":"Abs","activity":"Crunch","weights":10,"reps":12}`
  );
});
