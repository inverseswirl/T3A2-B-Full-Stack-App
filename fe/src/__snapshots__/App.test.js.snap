// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing renders correctly 1`] = `
<div
  className=""
>
  <div
    className="flex justify-between bg-sky-800"
  >
    <h1
      className="font-sans text-blue-200 basis-1/2 text-2xl sm:text-3xl md:text-4xl lg:text-6xl xl:text-7xl pl-3 py-1 xl:pl-12"
      data-testid="brandname"
    >
      Xero fitness
    </h1>
    <button
      className="pr-3 cursor-pointer"
      onClick={[Function]}
    >
      <svg
        color="rgb(191 219 254)"
        fill="currentColor"
        height={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "rgb(191 219 254)",
          }
        }
        viewBox="0 0 512 512"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
    </button>
  </div>
  <div
    className="flex flex-row justify-center"
  >
    <a
      className="bg-sky-800 text-blue-200 rounded-full px-16 py-2  mt-20 my-auto "
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      className="bg-sky-800 text-blue-200 rounded-full px-14 py-2  mt-20 ml-5 my-auto "
      href="/signup"
      onClick={[Function]}
    >
      Sign up
    </a>
  </div>
  <div
    className="bg-sky-800  border-t border-blue-300 pb-1 text-center text-blue-200 w-screen opacity-80 relative bottom-0"
  >
    <footer>
      Â© 2022
    </footer>
  </div>
</div>
`;

exports[`snapshot testing renders correctly 2`] = `
Array [
  <p
    className="text-blue-200 text-center text-xl border-b border-sky-800 pt-10"
  >
    Weights
  </p>,
  <div
    className="grid grid-cols-3 justify-items-center items-center"
  >
    <button
      className="mr-1 shadow-xl shadow-sky-900 hover:bg-sky-700 text-blue-200 bg-sky-800 opacity-90 rounded-full w-12 h-12 text-4xl cursor-pointer my-8 mx-auto"
      onClick={[Function]}
    >
       
      -
       
    </button>
    <input
      className="w-20 text-center focus:outline-none border-b border-blue-200  bg-transparent placeholder:italic placeholder:text-slate-400 text-blue-200"
      onChange={[Function]}
      type="number"
      value=""
    />
    <button
      className="ml-1 shadow-xl shadow-sky-900 hover:bg-sky-700 text-blue-200 bg-sky-800 opacity-90 rounded-full w-12 h-12 text-3xl cursor-pointer my-8 mx-auto"
      onClick={[Function]}
    >
      +
    </button>
  </div>,
  <p
    className="text-blue-200 text-center text-xl border-b border-sky-800 pt-10"
  >
    Reps
  </p>,
  <div
    className="grid grid-cols-3 justify-items-center items-center"
  >
    <button
      className="mr-1 shadow-xl shadow-sky-900 text-blue-200 hover:bg-sky-700 bg-sky-800 opacity-90 rounded-full w-12 h-12 text-4xl cursor-pointer my-8 mx-auto"
      onClick={[Function]}
    >
       
      -
       
    </button>
    <input
      className="w-20 text-center focus:outline-none border-b border-blue-200  bg-transparent placeholder:italic placeholder:text-slate-400 text-blue-200"
      onChange={[Function]}
      step="10"
      type="number"
      value=""
    />
    <button
      className="ml-1 shadow-xl shadow-sky-900 text-blue-200 hover:bg-sky-700 bg-sky-800 opacity-90 rounded-full w-12 h-12 text-3xl cursor-pointer my-8 mx-auto"
      onClick={[Function]}
    >
      +
    </button>
  </div>,
  <div
    className="flex flex-row basis-1/2 mt-5"
  >
    <form
      className="flex flex-row basis-1/2 "
      onSubmit={[Function]}
    >
      <input
        className="shadow-xl shadow-sky-900 text-blue-200 hover:bg-sky-700 bg-sky-800 opacity-85 rounded-sm w-20 h-14 text-2xl cursor-pointer my-6 mx-auto px-2"
        data-testid="save-button"
        type="submit"
        value="Save"
      />
    </form>
    <button
      className="shadow-xl shadow-sky-900 hover:bg-sky-700 text-blue-200 bg-sky-800 opacity-85 rounded-sm w-20 h-14 text-2xl cursor-pointer my-6 mx-auto px-2"
      onClick={[Function]}
    >
      Clear
    </button>
  </div>,
]
`;
